app/Http/Controllers/BalanceItemsController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\BalanceItem;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+
+class BalanceItemsController extends Controller
+{
+    public function show()
+    {
+        $user_id = Auth::id();
+
+        //get all items for the user
+        $items = BalanceItem::select(DB::raw('id, label, entry_date, (amount / 100) as amount, DATE(entry_date) as date'))
+                              ->orderBy('entry_date', 'desc')
+                              ->where('user_id', '=', $user_id)
+                              ->get();
```

Comment:
Did you consider using model relationships here to retrieve the user's balance items?

Reply:
I had not when I wrote this but that is a good thought. 

I will add the relationships in the models:

 User model:

public function balanceitems(){
     return $this->hasMany(BalanceItem::class);
}

BalanceItem model:

public function user(){
    return $this->belongsTo(User::class);
}

And then adjust the query to use these relationships instead.